i18n-helpers/fuzz/src/lib.rs

```diff
@@ -0,0 +1,49 @@
+use arbitrary::Arbitrary;
+use mdbook::book::{Book, BookItem, Chapter};
+use polib::catalog::Catalog;
+use polib::message::Message;
+use polib::metadata::CatalogMetadata;
+use std::path::PathBuf;
+
+/// Generate a random Catalog for fuzzing.
+pub fn create_catalog(translations: Vec<(&str, &str)>) -> Catalog {
+    let mut catalog = Catalog::new(CatalogMetadata::new());
+    for (idx, (msgid, msgstr)) in translations.iter().enumerate() {
+        let message = Message::build_singular()
+            .with_source(format!("foo.md:{idx}"))
+            .with_msgid(String::from(*msgid))
+            .with_msgstr(String::from(*msgstr))
+            .done();
+        catalog.append_or_update(message);
+    }
+    catalog
+}
+
+/// Generate a random Book for fuzzing.
+pub fn create_book(book_items: Vec<MyBookItem>) -> Book {
+    let mut book = Book::new();
+    book_items.into_iter().for_each(|item| {
```

Comment:
Please use a normal `for` loop — I don't really know why one would ever use `for_each` over a `for` loop.

Reply:
Done.